index.js

```
@@ -1,247 +1,28 @@
-        },
-        set: function(val) { _maxBandwidth = val }
-    });
-    Object.defineProperty(this, "minLatency", {
-        get: function() {
-            return Math.max(_minLatency, XHRShaper.minLatency);
-        },
-        set: function(val) { _minLatency = val }
-    });
-    Object.defineProperty(this, "minProgressEvents", {
-        get: function() {
-            return Math.max(_minProgressEvents, XHRShaper.minProgressEvents);
-        },
-        set: function(val) { _minProgressEvents = val }
-    });
-    Object.defineProperty(this, "randomness", {
-        get: function() {
-            return Math.max(_randomness, XHRShaper.randomness);
-        },
-        set: function(val) { _randomness = val }
-    });
-};
-
-XHRShaper.maxBandwidth = Infinity;
-XHRShaper.minLatency = 0;
-XHRShaper.minProgressEvents = 0;
-XHRShaper.randomness = 0;
-
-var WindowXHR = window.XMLHttpRequest;
-
-function mirrorReadOnlyProp(target, source, prop) {
-    Object.defineProperty(target, prop, {
-        get: function() {
-            return source[prop];
-        }
-    });
-}
-
-function mirrorRwProp(target, source, prop) {
-    Object.defineProperty(target, prop, {
-        get: function() {
-            return source[prop];
-        },
-        set: function(val) {
-            source[prop] = val;
-        }
-    });
+// Export all our stuff via module if we are one
+if (typeof module !== 'undefined') {
+    module.exports = {
+        BaseXHR: BaseXHR,
+        XMLHttpRequest: XMLHttpRequestShim,
+        useGlobal: useGlobal,
+        objectMirrors: objectMirrors
+    };
 }
 
-function mirrorFunc(target, source, func) {
-    //if (!source[func]) return; // in case someone doesn't implement the full standard bind() will fail here
-    target[func] = source[func].bind(source);
+function useGlobal() {
+    // Shim window/global XHR
+    var global = window || global;
+    if (typeof global !== 'undefined') {
```

Doesn't `window || global` cover all use cases? And what happens if global is undefined? Seems we just fail silently